src/Configuration/Programmer/Programmer_cmd.cpp

```diff
@@ -19,57 +19,145 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <iostream>
+
 #include "CFGCommon/CFGCommon.h"
 #include "Programmer.h"
 
+void progressCallback(std::string msg) { std::cout << msg; }
+
+void messageCallback(std::string msg) { std::cout << "MS-CALLBACK :" << msg; }
+
 using namespace FOEDAG;
 int main(int argc, const char** argv) {
   CFG_POST_MSG("This is Programmer cmd");
   CFGCommon_ARG cmdarg;
 
   // programmer_entry(&cmdarg);
-
   // simple programmer API manual testing
+
+  auto formatMemorySize = [](int sizeInBits) -> std::string {
+    const int bitsInKilobit = 1024;
+    const int bitsInMegabit = 1024 * 1024;
+
+    if (sizeInBits >= bitsInMegabit) {
+      int megabits = sizeInBits / bitsInMegabit;
+      return std::to_string(megabits) + "M";
+    } else if (sizeInBits >= bitsInKilobit) {
+      int kilobits = sizeInBits / bitsInKilobit;
+      return std::to_string(kilobits) + "K";
+    } else {
+      return std::to_string(sizeInBits);
+    }
+  };
+
+  std::string openOCDPath = "/home/guanyung/dev/openocd/src/openocd";
+  openOCDPath = "/home/asic01/development/openocd_rs/src/openocd";
```

Comment:
Why there is hardcoded path here? Or is it just temporarily?

Reply:
It is just a temporary thing. The openocd path should come from the user to set we programmer_cmd launch.